### PR TITLE
Add fraud detection tutorial

### PR DESCRIPTION
**Pull request by samfader**
## What?

This PR adds a new [tutorial](https://radar.com/documentation/tutorials) showing how to detect proxying and location mocking with the iOS SDK.

## Why?

Fraud detection was introduced in recent versions of the Radar SDK and is a use case that should be documented.

## How?

n/a

## Screenshots (optional)

image

### CODE REVIEW

File: docs/tutorials/detecting-location-based-fraud.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+### (Optional) Step 4: Use Radar Regions to perform a third location check
+
+ Radar's [Regions](/regions) functionality can be used to determine a user's regional context. Let's say your application requires a user to be in a specific city or region to unlock certain app functionality. An example solution would be to first check for proxying and mocking, and then as a third check, verify the user's regional location allows them to be eligible for your app's services:
+
+```swift
+func didUpdateLocation(_ location: CLLocation, user: RadarUser) {
+    // check to ensure a user is neither proxying nor mocking
+    if let user = user {
+        if (!user.mocked && !user.proxy){
```
Comment (jrtibbetts) [inline docs/tutorials/detecting-location-based-fraud.md:86]: `) {`. Better yet, drop the parentheses entirely.

File: docs/tutorials/detecting-location-based-fraud.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+If you haven't already, sign up for Radar to get your API key. You can create up to 1,000 geofences and make up to 100,000 API requests per month for free.
+
+<a className="btn btn-large btn-primary" href="https://radar.com/signup">Get API Keys</a>
+
+### Step 2: Install the Radar iOS SDK
+
+If you're starting from scratch, create a new Xcode project of type Single View App.
+
+[Install the Radar SDK](/sdk) using CocoaPods or Carthage (recommended) or by [downloading the framework](https://github.com/radarlabs/radar-sdk-ios/releases) and dragging it into your project.
```
Comment (jrtibbetts) [inline docs/tutorials/detecting-location-based-fraud.md:28]: Or Swift Package Manager!

Do we usually recommend Carthage instead of the others?

**Review by jrtibbetts — APPROVED** [review]
If checking for both `.fraud` and `.proxy` is a common use case, maybe we should have a single property that checks both.

File: docs/tutorials/detecting-location-based-fraud.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+}
+```
+
+If monitoring location in the background (i.e. using `Radar.startTracking()`): 
+
+```swift
+func didUpdateLocation(_ location: CLLocation, user: RadarUser) {
+    // check to ensure a user is neither proxying nor mocking
+    if let user = user {
+        if (!user.mocked && !user.proxy){
```
Comment (jrtibbetts) [inline docs/tutorials/detecting-location-based-fraud.md:69]: `) {`. Better yet, drop the parentheses entirely.

File: docs/tutorials/detecting-location-based-fraud.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+```
+
+### (Optional) Step 4: Use Radar Regions to perform a third location check
+
+ Radar's [Regions](/regions) functionality can be used to determine a user's regional context. Let's say your application requires a user to be in a specific city or region to unlock certain app functionality. An example solution would be to first check for proxying and mocking, and then as a third check, verify the user's regional location allows them to be eligible for your app's services:
+
+```swift
+func didUpdateLocation(_ location: CLLocation, user: RadarUser) {
+    // check to ensure a user is neither proxying nor mocking
+    if let user = user {
```
Comment (jrtibbetts) [inline docs/tutorials/detecting-location-based-fraud.md:85]: Same as in Step 3.

File: docs/tutorials/detecting-location-based-fraud.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    }   
+}
+```
+
+If monitoring location in the background (i.e. using `Radar.startTracking()`): 
+
+```swift
+func didUpdateLocation(_ location: CLLocation, user: RadarUser) {
+    // check to ensure a user is neither proxying nor mocking
+    if let user = user {
```
Comment (jrtibbetts) [inline docs/tutorials/detecting-location-based-fraud.md:68]: The `user` argument is non-Optional here, so you don't need this check.

**Review by jrtibbetts — APPROVED** [review]

File: docs/tutorials/detecting-location-based-fraud.md (hunk 2 above)
Comment (samfader) [reply docs/tutorials/detecting-location-based-fraud.md:28]: Ah, I just copied this section from other tutorials. They really all need to be updated to be more accurate.

File: docs/tutorials/detecting-location-based-fraud.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            // user is spoofing location or on a proxy. Message about block.
+        }
+    }   
+}
+```
+
+If monitoring location in the background (i.e. using `Radar.startTracking()`): 
+
+```swift
+func didUpdateLocation(_ location: CLLocation, user: RadarUser) {
```
[the remark this answers is not in the document]
Comment (samfader) [reply docs/tutorials/detecting-location-based-fraud.md:66]: @jrtibbetts What do you think is the best way to do that? A comment or the full code block?

File: docs/tutorials.mdx [hunk 7]
```diff
@@ -15,6 +15,7 @@ Get to `Hello, world!` faster with tutorials and quickstarts! Or, get started wi
 - **[Building an app experience that changes based on location (e.g., in-store mode)](/tutorials/building-an-app-experience-that-changes-based-on-location-in-store-mode)**: Use the [iOS SDK](/sdk/ios) and [geofences](/geofences) to build a "RadarMart" app that defaults to an in-store mode when the user is in a store geofence.
 - **[Building a store locator on iOS](/tutorials/building-a-store-locator-on-ios)**: Use the [iOS SDK](/sdk/ios), [forward geocode API](/api#forward-geocode), and [geofence search API](/api#search-geofences) to build a store locator with a map.
 - **[Localizing a website based on current country or city](/tutorials/localizing-a-website-based-on-current-country-or-city)**: Use the [web SDK](/sdk/web) and [IP geocoding API](/api#ip-geocode) to localize a website based on current country or city.
+- **[Detecting location-based fraud](/tutorials/detecting-location-based-fraud)**: Use the [web SDK](/sdk/web) and [IP geocoding API](/api#ip-geocode) to localize a website based on current country or city.
```
Comment (matthewisabel) [inline docs/tutorials.mdx:18]: Needs new description

**Pull request merged**